### ISSUE TITLE
Bug: [no-unused-vars] False negative when var is referenced from TSDoc `{@link ...}` / tsdoc support

### ISSUE THREAD

**Issue by kraenhansen** [typescript-eslint/typescript-eslint#6967]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.0.4&sourceType=module&code=JYWwDg9gTgLgBAbzgZQiApjAFsAdgczgF84AzKNOAcgGc10B3LdKdKgbgCgB6AKl85xecAKo10NONnSIAAgBs8AaxT1sefEUG9undAA9IsOAGN5AQxqSAwhFw0ArhiiJinIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tidmUQAmtAG4BDaKgyRE0aB2iRwYAL4hlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
import { Something } from 'somewhere';
/**
 * Uses the {@link Something}
 */
export class Consumer { }
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-unused-vars": "error"
  }
};
```

### Expected Result

I expected the `{@link Something}` to mark `Something` as used and therefore not fail on line 1

### Actual Result

An error on line 1:
> 'Something' is defined but never used. 1:10 - 1:19

### Additional Info

When I <kbd>Ctrl</kbd> + click in VSCode my editor jumps to the class definition as expected and [TypeDoc](https://typedoc.org/) will generate links when mentioning a class like this, but it must be imported.

This is a link for the `{@link}` inline tag: https://tsdoc.org/pages/tags/link/

**Comment by kraenhansen** [comment]
Related reading:
- https://eslint.org/blog/2018/11/jsdoc-end-of-life/ (article on why non eslint doesn't support it)
- https://github.com/gajus/eslint-plugin-jsdoc/issues/858 (a similar issue reported with eslint-plugin-jsdoc)

**Comment by kraenhansen** [comment]
I did search for other issues mentioning this, but apparently not well enough as I see my issue is a duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/5017, which was closed as a wontfix and reference the eslint-pligin-jsdoc that I did manage to find. Anyways - I'll close this again and continue in that other issue.
